components/lib/n3-utils.js

```diff
@@ -0,0 +1,291 @@
+                            object: toN3(triple.object),
+                            graph: graphName == '@default' ? undefined : graphName
+                        });
+
+                    });
+            });
+
+            return inGraph;
+
+    }).catch(function(e) {
+        logger.error("Exception converting JSON-LD to N3:", util.inspect(e, {depth:null}));
+        throw Error("Unable to convert JSON-LD to N3: " + e.message);
+    });
+}
+
+/** 
+ * Converts an RDF element to the appropriate N3 value or literal
+ *  
+ * @param term the triple element to process
+ *
+ * @return the correct N3 literal or value
+ */
+function toN3(term) {
+
+    if (term.type != 'literal') {
+        return term.value;
+    }
+
+    return N3.Util.createLiteral(term.value, 
+                                 term.language || term.datatype);
+}
+
+/**
+ * Converts an n3 graph to ttl and writes it to a file if one is specified
+ * 
+ * @param store    (required) the N3 store to convert to text/Turtle format
+ * @param filename (optional) an optional file name to write the Turtle result into.
+ *
+ * @return a promise to return the N3 store converted to Turtle format or a reject on error
+ */
+function n3ToTtl(store, filename) {
+
+    if (_.isEmpty(store)) { 
+        return Promise.reject(errUtils.errorMessage("n3ToTtl called with an empty N3 store!"));
+    }
+
+    return new Promise(
+        function(resolve, reject) { 
+            var writer = N3.Writer(); 
+
+            store.find().forEach(t => { 
+                writer.addTriple(t); 
+            });
+
+            writer.end(function(error, result) {
+
+                if (!_.isNull(error)) { 
+                    return reject(errUtils.errorMessage("Unable to convert N3 store to TTL!", error));  
+                }  
+    
+                if (_.isEmpty(filename)) { 
+                    return resolve(result);
+                }
+
+                return fs.writeFile(filename, result, function(error) { 
+                    if (!_.isNull(error)) { 
+                        return reject(errUtils.errorMessage("n3ToTtl API was unable to write file" + filename + "!", error));
+                    } 
+
+                    return resolve(result);
+                });
+
+            });
+
+        }).catch(function(e) { 
+            return Promise.reject(errUtils.errorMessage('Unable to convert N3 store to TTL!',e));
+        });
+}
+
+/**
+ * Converts an N3 store into TTL, loads the TTL, and converts the TTL to JSON-LD.  This function
+ * is intended primarily for testing/double checking the n3ToJsonld method and will likely be 
+ * deleted or deprecated in the future. 
+ *
+ * @param store    (required) the N3 store to convert to text/Turtle format
+ * @param frame    (optional) a JSON-LD frame
+ * @param filename (optional) an optional file name to write the Turtle result into.
+ *
+ * @return a promise to return the N3 store converted to JSON-LD format or a reject on error
```

Comment:
Suggest rewording this: @return a promise to return either the JSON-LD format or a reject on error